workbench-core/base/src/aws/helpers/s3Service.ts

```diff
@@ -75,4 +78,52 @@ export default class S3Service {
 
     await recursiveUpload(path, '');
   }
+
+  /**
+   * Read Template from Bucket
+   * @param s3BucketURL - URL of provision artifact template
+   * @returns json object containing yaml file configuration
+   */
+  public async getTemplateByURL(s3BucketURL: string): Promise<{
+    Parameters: {
+      [key: string]: {
+        Type: string;
+        Default?: string;
+        AllowedValues?: string[];
+        Description: string;
+      };
+    };
+  }> {
+    const s3BucketParams = s3BucketURL.split('.s3.amazonaws.com/');
+    if (s3BucketParams.length !== 2) throw new Error(`Invalid S3 URL format ${s3BucketURL}`);
+    const s3Bucket = s3BucketParams[0].replace('https://', '');
+    const key = s3BucketParams[1];
+    const stream = await this._s3.getObject({ Bucket: s3Bucket, Key: key });
+    const streamString = await this._streamToString(stream.Body! as Readable);
+    const yamlFile = await yaml.load(streamString, { schema: schema });
+    return yamlFile as {
```

Comment:
Is there a step in here where we need to pull only the `Parameters` section? Doesn't `yamlFile` include the whole CFN template. I see you're setting the type here to only include `Parameters`. However, where is the step to convert `yamlFile` to only include `Parameters`?

Reply:
I decided to retrieve all the template object in case we need something else later coming from there,
in `environmentTypeHandler` at the moment of saving im only calling object.Parameters 
```
 await envTypeService.createNewEnvironmentType({
      productId: product.ProductId || '',
      provisioningArtifactId: provisionArtifact.Id || '',
      description: provisionArtifact.Description || '',
      name: `${product.Name}-${provisionArtifact.Name}`,
      type: product.Name || '',
      params: provisionParams.Parameters,
      status: 'NOT_APPROVED'
    }); 
```
Do you think is overhead to retrieve everything and I should get only the params in the s3Service ?

Reply:
Ok, that makes sense, but why does the `yamlFile` definition only have this
```
Parameters: {
        [key: string]: {
          Type: string;
          Default?: string;
          AllowedValues?: string[];
          Description: string;
        };
      };
     ```
     
Isn't there also a Condition and Resource section? The `type` defined for `yamlFile` confuses me.

Reply:
updated to use interface